addons/account/models/account_move_line.py

```diff
@@ -2453,7 +2453,8 @@ def _prepare_analytic_lines(self):
 
             for account_id, distribution in self.analytic_distribution.items():
                 line_values = self._prepare_analytic_distribution_line(float(distribution), account_id, distribution_on_each_plan)
-                if not float_is_zero(line_values.get("amount"), precision_digits=self.env.company.currency_id.decimal_places):
+                comp_curr = self.company_currency_id or self.env.company.currency_id
```

Comment:
When can `self.company_currency_id` not be set when computing the analytic values?

The only possible case I see is in the `_inverse_analytic_distribution` because of the `@api.onchange` but I don't think it makes sense to have it since we don't need the value dynamically in the form view.

Reply:
I don't even know why I used the `company_currency_id` instead of the aml's `currency_id`